soe/dynamic_test.go

```diff
@@ -0,0 +1,206 @@
+package soe_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/hamba/avro/v2"
+	"github.com/hamba/avro/v2/soe"
+	"github.com/hamba/avro/v2/soe/testdata"
+	"github.com/stretchr/testify/require"
+)
+
+// Helper function to marshal v into an SOE-framed binary Avro encoding.
+func encode(t *testing.T, schema avro.Schema, v any) []byte {
+	t.Helper()
+
+	// Use a basic codec to marshal value.
+	codec, err := soe.NewCodec(schema)
+	require.NoError(t, err)
+
+	data, err := codec.Encode(&v)
+	require.NoError(t, err)
+	return data
+}
+
+// Helper function to create a new DynamicDecoder with a registry over zero or
+// more schemas.
+func newDynamicDecoder(t *testing.T, schemas ...avro.Schema) *soe.DynamicDecoder {
+	t.Helper()
+
+	// Set up a registry with provided schemas.
+	registry := soe.NewMemorySchemaRegistry()
+	for _, schema := range schemas {
+		err := registry.AddSchema(schema)
+		require.NoError(t, err)
+	}
+	return soe.NewDynamicDecoder(registry)
+}
+
+func TestDynamicDecodeUnknownSchema(t *testing.T) {
```

Comment:
Should probably use the `t.Run()` pattern to avoid having to prefix every test function with distinct `TestDynamic`/`TestTyped` prefixes.

Reply:
This is not necessary better. It depends on the test setup. `t.Run`/table tests syntax works well when the test setup is identical. In this case, the test setup varies some, so IMO it is better the way you have it.

Reply:
Used `t.Run` in some cases where I do share setup.